HACKING.md

```diff
@@ -50,3 +52,106 @@ Overview of Mender client architecture.
 +------------------+       +------------------------+        +--------------------+
 
 ```
+
+## Hacking on Mender client locally
+
+Build mender with support for local development:
+```
+$ make LOCAL=1
+```
+
+Next step is to create required prefix tree that looks similar to what is present on the device. We want to reach a tree like this:
+
+```
+prefix (could be /usr, or other)
+├── [drwxr-xr-x]  etc
+│   └── [drwxr-xr-x]  mender
+│       └── [-rw-r--r--]  mender.conf
+├── [drwxr-xr-x]  share
+│   └── [drwxr-xr-x]  mender
+│       ├── [drwxr-xr-x]  identity
+│       │   └── [-rwxr-xr-x]  mender-device-identity
+│       └── [drwxr-xr-x]  inventory
+│           ├── [-rwxr-xr-x]  mender-inventory-hostinfo
+│           └── [-rwxr-xr-x]  mender-inventory-network
+└── [drwxr-xr-x]  var
+    └── [drwxr-xr-x]  lib
+        └── [drwxr-xr-x]  mender
+            └── [-rw-r--r--]  authtentoken
+```
+
+First, create necessary directories:
+
+```
+$ mkdir -p $PWD/prefix/etc/mender \
+    $PWD/prefix/share/mender/{identity,inventory} \
+    $PWD/prefix/var/lib/mender
+```
+
+Provide a dummy tenant token:
+
+```
+$ echo dummy > $PWD/prefix/var/lib/mender/authtentoken
+```
+
+Use example config file as template (edit as needed, ex. point the client to
+http://docker.mender.io:9080):
+
+```
+$ cp mender.conf.example $PWD/prefix/etc/mender/mender.conf
+```
+
+Use example device identity script:
```

Comment:
Mentioning authentication without explaining how it works in the system is pointless. For all the reader knows, the script provides device identity.